torch/csrc/jit/passes/onnx.cpp

```diff
@@ -169,8 +169,14 @@ std::shared_ptr<Graph> ToONNX(
   ConstantValueMap::ClearMaps();
   auto new_graph = std::make_shared<Graph>(graph->current_scope());
   std::unordered_map<Value*, Value*> env;
+  py::set values_in_env;
```

Comment:
Could you add the rationale as a comment here for future readers? Thanks!

Reply:
Added a comment, thanks! I also pushed a rebase since #123063 merged, that touches many of the same lines.